stroyprombeton/admin/filters.py

```diff
@@ -49,25 +53,23 @@ def find_duplicates_by_field(queryset, field_name):
                 .order_by()
                 .filter(c__gt=1)
                 .values_list('ids', flat=True)
-            ))
 
-        if self.value() == 'h1':
+        if lookup == 'markh1':
             ids = [
                 find_duplicates_by_field(queryset, field_name)
-                for field_name in ['name', 'stroyprombeton_product__mark']
+                for field_name in ['product__page__name', 'mark']
+
             ]
             return queryset.filter(id__in=set.intersection(*ids))
 
-        if self.value() == 'all':
+        if lookup == 'all':
             ids = [
                 find_duplicates_by_field(queryset, field_name)
                 for field_name in [
-                    'stroyprombeton_product__price', 'stroyprombeton_product__code',
-                    'stroyprombeton_product__mark', 'stroyprombeton_product__specification',
-                    'stroyprombeton_product__length', 'stroyprombeton_product__width',
-                    'stroyprombeton_product__height', 'stroyprombeton_product__weight',
-                    'stroyprombeton_product__volume', 'stroyprombeton_product__diameter_out',
-                    'stroyprombeton_product__diameter_in', 'description', 'title', 'name', 'h1'
+                    'price', 'code',
+                    'mark', 'product__category__specification',
```

Comment:
```suggestion
                    'mark',
```

all specification fields will be moved to Series separated model